packages/core/src/components/Autocomplete/Autocomplete.jsx

```diff
@@ -0,0 +1,174 @@
+
+    this.id = uniqueId('autocomplete_');
+    this.labelId = uniqueId('constrained-list-header_');
+  }
+
+  renderChildren(getInputProps) {
+    // Extend props on the TextField, by passing them through
+    // Downshift's `getInputProps` method
+    return React.Children.map(this.props.children, child => {
+      if (isTextField(child)) {
+        return React.cloneElement(
+          child,
+          getInputProps({
+            fieldClassName: 'ds-c-autocomplete__input',
+            id: this.id,
+            labelClassName: 'ds-u-margin-top--0 ds-c-autocomplete__label'
+          })
+        );
+      }
+
+      return child;
+    });
+  }
+
+  render() {
+    const {
+      clearAriaLabel,
+      clearInputText,
+      items,
+      itemToString,
+      label,
+      onChange
+    } = this.props;
+
+    return (
+      <Downshift
+        itemToString={itemToString}
+        onChange={onChange}
+        render={({
+          clearSelection,
+          getInputProps,
+          getItemProps,
+          highlightedIndex,
+          isOpen,
+          inputValue
+        }) => (
+          <div className="ds-u-clearfix ds-c-autocomplete">
+            {this.renderChildren(getInputProps)}
+
+            {isOpen ? (
+              <div className="ds-u-border--1 ds-u-padding--1 ds-c-autocomplete__list">
+                {label && (
+                  <h5
+                    className="ds-u-margin--0 ds-u-padding--1 ds-u-padding-left--2"
+                    id={this.labelId}
+                  >
+                    {label}
+                  </h5>
+                )}
+
+                <ul
+                  aria-labelledby={label ? this.labelId : null}
+                  className="ds-c-list--bare"
+                  role="listbox"
+                >
+                  {items
+                    .filter(
+                      item =>
+                        !inputValue ||
+                        item.name
+                          .toLowerCase()
+                          .includes(inputValue.toLowerCase())
+                    )
+                    .map((item, index) => (
+                      <li
+                        aria-selected={highlightedIndex === index}
+                        className={
+                          highlightedIndex === index
+                            ? 'ds-c-autocomplete__list-item ds-c-autocomplete__list-item--active'
+                            : 'ds-c-autocomplete__list-item'
+                        }
+                        key={item.id}
+                        role="option"
+                        {...getItemProps({ item })}
+                      >
+                        {itemToString(item)}
+                      </li>
+                    ))}
+                </ul>
+              </div>
+            ) : null}
+
+            <Button
+              aria-label={clearAriaLabel}
+              className="ds-u-float--right ds-u-padding-right--0"
+              href="javascript:void(0);"
```

Comment:
@1Copenut In a previous comment you said:

>  forcing it to render an `<a>` tag. It was either a link, or a lot of overriding to get the button to not have a background when disabled. Will change as needed.

Now that we're no longer passing in the disabled state into `<Autocomplete>`, I'd push for us to remove the `href` and use the more semantically correct `<button>` tag for this action. 

(I'd argue that anywhere you're using `javascript:void(0)` as the `href`, then the element should be a `button`).

Reply:
Done. Converted to a button. This is better semantics for sure.